### PR TITLE
Autotests: #3186 – rectangle selection tool

### PR DESCRIPTION
**Pull request by y-holik**
## How the feature works? / How did you fix the issue?
(Screenshots, videos, or GIFs, if applicable)

## Check list
- [ ] unit-tests written
- [x] e2e-tests written
- [x] documentation updated
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name doesn't contain '#'
- [x] PR is linked with the issue
- [x] base branch (master or release/xx) is correct
- [x] task status changed to "Code review"
- [x] reviewers are notified about the pull request

### CODE REVIEW

File: ketcher-autotests/tests/test-data/Rxn-V2000/benzene-chain-reaction.rxn [hunk 1]
```diff
@@ -68,4 +68,8 @@ $MOL
 
   1  0  0     0  0            999 V2000
    25.8000   -6.3500    0.0000 Br  0  0  0  0  0  0  0  0  0  0  0  0
+<<<<<<< HEAD
```
Comment (AKZhuk) [inline ketcher-autotests/tests/test-data/Rxn-V2000/benzene-chain-reaction.rxn:71]: It looks like unresolved merge conflict.. Is it expected?

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+  const xMark = 300;
+  const yMark = 200;
+
+  async function selectObjects(page: Page, xAxis: number, yAxis: number) {
+    const point = await getCoordinatesOfTheMiddleOfTheScreen(page);
+    await page.mouse.move(point.x - xAxis, point.y - yAxis);
+    await page.mouse.down();
+    await page.mouse.move(point.x + xAxis, point.y + yAxis);
+    await page.mouse.up();
```
Comment (AKZhuk) [inline ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts:43]: I guess you can use function dragMouseTo

Comment (y-holik) [reply ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts:43]: when I use dragMouseTo something goes terribly wrong. Is it ok if I leave it as page.mouse.move?

File: ketcher-autotests/tests/test-data/Rxn-V2000/benzene-chain-reaction.rxn (hunk 1 above)
Comment (y-holik) [reply ketcher-autotests/tests/test-data/Rxn-V2000/benzene-chain-reaction.rxn:71]: resolved

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts (hunk 2 above)
Comment (AKZhuk) [reply ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts:43]: sure

**Review by AKZhuk — APPROVED** [review]

**Review by AnastasiyaPiatrovaKlu — APPROVED** [review]

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    const atomPoint = await getAtomByIndex(page, { label: 'C' }, atomLabel);
+    await dragMouseTo(atomPoint.x, atomPoint.y, page);
+  });
+
+  test('Fusing bonds together', async ({ page }) => {
+    //  Test case: EPMLSOPKET-1351
+    const moveMouseCoordinatesY = 10;
+    const objectSelection = 50;
+    await drawBenzeneRing(page);
+    await selectNestedTool(page, BondTool.SINGLE_AROMATIC);
```
Comment (Nitvex) [inline ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts:132]: can we use `selectDropdownTool`?

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+  test.afterEach(async ({ page }) => {
+    await takeEditorScreenshot(page);
+  });
+
+  const xDelta = 30;
+  const yDelta = 60;
+  const modifier = getControlModifier();
+
+  const xMark = 300;
```
Comment (Nitvex) [inline ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts:34]: may be we can use:
```
const emptySpace = { x: 300, y: 200 };
async function clickCanvas(page: Page) {
    await page.mouse.click(emptySpace.x, emptySpace.y);
  }
```

Did I get your point correctly?

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  const yMark = 200;
+
+  async function selectObjects(page: Page, xAxis: number, yAxis: number) {
```
Comment (Nitvex) [inline ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts:37]: May be we can use something like `xAxisRadius` or something?

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+    const moveMouseCoordinatesY = 10;
+    const moveMouseCoordinatesX = 270;
+    await openFileAndAddToCanvas('Rxn-V2000/benzene-chain-reaction.rxn', page);
+    await page.getByTestId('select-rectangle').click();
+    const point = await getCoordinatesOfTheMiddleOfTheScreen(page);
+    await page.mouse.move(
+      point.x - moveMouseCoordinatesX,
+      point.y + moveMouseCoordinatesY,
+    );
+    await page.mouse.down();
```
Comment (Nitvex) [inline ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts:83]: Why do we need mouse.down and then mouse.up? => it is a click
And then we again perform a click?

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+  test('Fusing bonds together', async ({ page }) => {
+    //  Test case: EPMLSOPKET-1351
+    const moveMouseCoordinatesY = 10;
+    const objectSelection = 50;
+    await drawBenzeneRing(page);
+    await selectNestedTool(page, BondTool.SINGLE_AROMATIC);
+    const coordinates = await getCoordinatesTopAtomOfBenzeneRing(page);
+    await page.mouse.click(coordinates.x + xDelta, coordinates.y - yDelta);
+    await page.getByTestId('select-rectangle').click();
```
Comment (Nitvex) [inline ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts:135]: Suggestion – your test looks overloaded. If you perform multiple actions like click here, then there, then drag and so on, then create a function for this. 
This is not about code duplication, this is about meaning. Just imagine, how would you describe your test. You don't tell people:
1) Move mouse here
2) Then click in this coordinates
3) Then drag

You tell – select structure. Of course, you can skip this step, when your test is 5 lines long. But when you have a lot of clicks, drags, mouse moves in your tests it becomes hard to read. So ideally, your test should look like:
```
await selectRectangleSelection();
await selectBenzeneRing();
await moveBenzeneRing();
...
```

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+    await selectObjects(page, objectSelection, objectSelection);
+    const atomNumber = 4;
+    const bondPoint = await getBondByIndex(page, {}, atomNumber);
```
Comment (Nitvex) [inline ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts:138]: Why `atomNumber` and `getBondByIndex(page, {}, atomNumber)`?

**Review by Nitvex — CHANGES_REQUESTED** [review]

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts (hunk 3 above)
Comment (y-holik) [reply ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts:132]: done

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts (hunk 5 above)
Comment (y-holik) [reply ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts:37]: done

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts (hunk 6 above)
Comment (y-holik) [reply ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts:83]: done

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts (hunk 7 above)
Comment (y-holik) [reply ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts:135]: done

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts (hunk 8 above)
Comment (y-holik) [reply ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts:138]: fixed

**Review by Nitvex — APPROVED** [review]

**Pull request merged**